### PR TITLE
Bump to v0.9.38

### PR DESCRIPTION
**Pull request by yrong**
- [x] upgrade dependencies & make code compiled
- [x] refactoring for [#12345 pallet-mmr: RPC API and Runtime API work with block numbers](https://github.com/paritytech/substrate/pull/12345) and [#12530](https://github.com/paritytech/substrate/pull/12530)

companion with https://github.com/Snowfork/go-substrate-rpc-client/pull/57

also need companion with https://github.com/Snowfork/snowbridge/pull/780 though we break it for easy review or the beefy bridge will not work

### CODE REVIEW

File: parachain/Cargo.toml [hunk 1]
```diff
@@ -11,77 +11,77 @@ repository = "https://github.com/Snowfork/snowbridge"
...
+sc-cli = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-client-api = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-consensus = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-executor = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-keystore = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-network = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-rpc = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-rpc-api = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-sysinfo = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
+sc-service = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
```
[the remark this answers is not in the document]
Comment (yrong) [reply parachain/Cargo.toml:39]: removed in this [PR](https://github.com/paritytech/substrate/pull/12684)

File: parachain/pallets/ethereum-beacon-client/Cargo.toml [hunk 2]
```diff
@@ -21,24 +21,24 @@ byte-slice-cast = { version = "1.2.1", default-features = false }
...
 snowbridge-beacon-primitives = { path = "../../primitives/beacon", default-features = false }
 
 [dev-dependencies]
-sp-keyring = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.30" }
+sp-keyring = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
 snowbridge-testutils = { path = "../../primitives/testutils" }
 serde_json = "1.0.68"
 hex-literal = { version = "0.3.4" }
-pallet-timestamp = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.30", default-features = false }
+pallet-timestamp = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
```
[the remark this answers is not in the document]
Comment (yrong) [reply parachain/pallets/ethereum-beacon-client/Cargo.toml:41]: Actually the [TimeProvider](https://github.com/Snowfork/snowbridge/blob/a8564bb83f5838f486c867e1a57d46bb71225e61/parachain/pallets/ethereum-beacon-client/src/lib.rs#L458) we depend here is some trait so I would assume should be fine as dev dependency only

File: parachain/pallets/xcm-support/src/lib.rs [hunk 3]
```diff
@@ -85,7 +85,7 @@ pub mod pallet {
 			let origin_location: MultiLocation = MultiLocation {
 				parents: 0,
 				interior: Junctions::X1(Junction::AccountId32 {
-					network: NetworkId::Any,
+					network: None,
```
[the remark this answers is not in the document]
Comment (yrong) [reply parachain/pallets/xcm-support/src/lib.rs:88]: https://github.com/Snowfork/snowbridge/pull/775/commits/7c99eaa83d12ee04ab5a641d52041300e8ad3909
removed from runtime now and will do a final cleanup in another PR  with `ehtereum-light-client` together

File: .github/workflows/parachain.yml [hunk 4]
```diff
@@ -3,21 +3,21 @@ name: parachain
...
-      - '!parachain/LICENSE'
+      - "parachain/**"
+      - "!parachain/README.md"
+      - "!parachain/LICENSE"
   workflow_dispatch:
 
 jobs:
   fmt:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
Comment (vgeddes) [inline .github/workflows/parachain.yml:20]: I've setup a [larger github-hosted runner](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners) for our org, with 8 vcpus and 32GiB of RAM. Should help a lot to decreases execution time.

To use it:

```suggestion
    runs-on: snowbridge-runner
```

**Review by vgeddes — COMMENTED** [review]
Hey @yrong, we don't need really need the test-parachain under `parachain/utils/test-parachain` anymore, so there is no need to upgrade it.

Instead, what we will do is launch the Statemint parachain in our E2E testnet. But we can handle this in a future PR.

cc @alistair-singh This make sense to you?

**Comment by alistair-singh** [comment]
> Hey @yrong, we don't need really need the test-parachain under `parachain/utils/test-parachain` anymore, so there is no need to upgrade it.
> 
> Instead, what we will do is launch the Statemint parachain in our E2E testnet. But we can handle this in a future PR.
> 
> cc @alistair-singh This make sense to you?

I agree.

**Review by vgeddes — APPROVED** [review]
🚀

**Review by alistair-singh — APPROVED** [review]

**Pull request merged**